### ISSUE TITLE
Point not found in PolyhedronTrellis at certain q-points

### ISSUE THREAD

**Issue by rebeccafair** [brille/brille#61]
When I run the following script (very similar to the one in #60)
```
import spglib
import numpy as np
from euphonic import ureg, ForceConstants
import brille as br

fc_file = 'castep_files/quartz.castep_bin'
fc = ForceConstants.from_castep(fc_file)

crystal = fc.crystal
cell_vectors = crystal._cell_vectors
cell = crystal.to_spglib_cell()

dataset = spglib.get_symmetry_dataset(cell)
rotations = dataset['rotations']  # in fractional
translations = dataset['translations']  # in fractional

symmetry = br.Symmetry(rotations, translations)
direct = br.Direct(*cell)
direct.spacegroup = symmetry
bz = br.BrillouinZone(direct.star)

vol = bz.ir_polyhedron.volume
n_grid_points=2000
grid = br.BZTrellisQdc(bz, node_volume_fraction=vol/n_grid_points)

phonons = fc.calculate_qpoint_phonon_modes(grid.rlu)

evecs_basis = np.einsum('ba,ijkb->ijka', np.linalg.inv(cell_vectors),
                        phonons.eigenvectors)
n_atoms = crystal.n_atoms
frequencies = np.reshape(phonons._frequencies,
phonons._frequencies.shape + (1,))
freq_el = (1,)
freq_weight = (1., 0., 0.)
evecs = np.reshape(evecs_basis,
                   (evecs_basis.shape[0], 3*n_atoms, 3*n_atoms))
cost_function = 0
evecs_el = (0, 3*n_atoms, 0, 3, 0, cost_function)
evecs_weight = (0., 1., 0.)
grid.fill(frequencies, freq_el, freq_weight, evecs, evecs_el,
          evecs_weight)

qpts = np.array([[-0.1 + 1e-9, -0.1, 0.]])
vals, vecs = grid.ir_interpolate_at(qpts, useparallel=False)
```
I get the following error: `RuntimeError: Point not found in PolyhedronTrellis`. Specifically, this happens with `quartz.castep_bin`, `n_grid_points=2000` and `useparallel=False`. Note that the q-point it fails on is `[-0.1 + 1e-9, -0.1, 0.]`, `[-0.1, -0.1, 0.]` seems to work ok. I'm using `Brille 0.5.4` from PyPI on Windows.

**Comment by g5t** [comment]
The failure for `[-0.1 + 1e-9, -0.1, 0.]` but not `[-0.1, -0.1, 0.]` points towards a rounding problem, probably related to my implementation of floating point comparisons in `brille::approx::scalar`.

The `Point not found` error is thrown because the `PolyhedronTrellis` has identified that the test-point should be inside of a particular `Node` based on the Cartesian 'trellis' boundaries, but that node (a `PolyNode` in this case) found no vertices or weights required for interpolation when asked. Since each `PolyNode` is made up of a series of tetrahedra, it checks *which* tetrahedron contains the point by calculating the four interpolation weights -- if all weights are positive (with up to three equal to zero) then that tetrahedron contains the point.
In this specific case, one vertex of the containing tetrahedron requires an interpolation weight of ~`-1e-8` which `brille::approx::scalar` is set to indicate is not approximately zero; and therefore no tetrahedron vertices or weights are returned for the interpolation.

By changing the algorithm to *prefer* a fully-correct solution with no negative weights but to accept a least-negative solution if necessary this error is avoided.

Investigating the differences between `brillle::Polyhedron` inclusion checks and Tetgen's `orient3d` may yield a more robust solution.

**Comment by g5t** [comment]
This was fixed by #62 and is in the release `v0.5.7` and on [PyPI](https://pypi.org/project/brille/)
